components/rss/actions/merge-rss-feeds/merge-rss-feeds.mts

```diff
@@ -0,0 +1,34 @@
+import rss from "../../rss.app.mjs";
+import Parser from "rss-parser";
+import {
+  Action, UserProp,
+} from "@pipedream/types";
+
+export default {
+  name: "Merge RSS Feeds",
+  description: "Retrieve multiple RSS feeds and return a merged array of items sorted by date.",
+  key: "rss-merge-rss-feeds",
+  version: "0.1.{{ts}}",
+  type: "action",
+  props: {
+    feeds: {
+      type: "string[]",
+      label: "Feeds",
+      description: "The list of RSS feeds you want to parse.",
+    } as UserProp,
+    rss,
+  },
+  async run() {
+    const parser = new Parser();
+    const requests = this.feeds.map((feed: string) => parser.parseURL(feed));
+
+    const results = await Promise.all(requests);
+
+    const items = results.map((feed: Parser.Output<string>) => feed.items.map((item: object) => ({
+      feed,
+      item,
+    })));
+
+    return items;
+  },
+} as Action;
```

Comment:
I think the [`ironpipe`](https://github.com/ci7lus/ironpipe) library uses a `defineComponent()` function for typing rather than type assertion so the ThisType of the component's methods can be constrained using `defineComponent`'s first argument.
I could be mistaken, but I think if we use type assertion, we'd have to provide the `props` & `methods` types as generics to `Action<P,M>` in order to type `this`. For example:
```ts
export interface Action<M extends Methods, P> {
  methods?: M & ThisType<M & P>;
```

Is there are reason for not using and updating `ironpipe`'s `defineComponent`/`defineAction` for typing components?